kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/monitor/EthernetMonitorServiceImpl.java

```diff
@@ -309,6 +309,12 @@ private void monitor(String interfaceName) {
                             logger.debug("link is down - disabling {}", interfaceName);
                             disableInterface(interfaceName);
                             interfaceStateChanged = true;
+                        } else if (isDhcpClient && prevInterfaceState == null || currentInterfaceState
+                                .getCarrierChanges() != prevInterfaceState.getCarrierChanges()) {
```

Comment:
If the prevInterfaceState is null, the right part of the condition (after the ||) throws a NPE.